### ISSUE TITLE
Import error with message "external_id"

### ISSUE THREAD

**Issue by Darkheir** [OpenCTI-Platform/client-python#164]
## Description

During an import I have an error: `external_id`

![image](https://user-images.githubusercontent.com/679995/115599919-6bcc7b80-a2dc-11eb-8f91-e9f9eb113a25.png)

In the log I have only

```
INFO:root:Importing a attack-pattern (id: attack-pattern--6cd1a813-ccdf-4ba0-9b54-cb808f1059cc)
ERROR:root:'external_id'
```

The error looks like if we logged the string representation of a `KeyError`

I checked a bit and there is at least one place where we try to access `external_id` without checking its existence:

https://github.com/OpenCTI-Platform/client-python/blob/8c32efd38a4f24ec19ca28fa3c7f27014baf2104/pycti/utils/opencti_stix2.py#L966-L970

The bundle causing the error:

```json
{
    "type": "bundle",
    "id": "bundle--922a3b48-9c41-4d74-bafc-cc19e299924e",
    "spec_version": "2.1",
    "objects": [
        {
            "spec_version": "2.1",
            "x_ic_is_in_flint": true,
            "external_references": [
                {
                    "url": "https://attack.mitre.org/techniques/T1566/002",
                    "source_name": "mitre-attack"
                },
                {
                    "url": "https://capec.mitre.org/data/definitions/163.html",
                    "source_name": "capec"
                },
                {
                    "url": "https://blog.trendmicro.com/trendlabs-security-intelligence/pawn-storm-abuses-open-authentication-advanced-social-engineering-attacks",
                    "description": "Hacquebord, F.. (2017, April 25). Pawn Storm Abuses Open Authentication in Advanced Social Engineering Attacks. Retrieved October 4, 2019.",
                    "source_name": "Trend Micro Pawn Storm OAuth 2017"
                }
            ],
            "type": "attack-pattern",
            "x_ic_deprecated": false,
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ],
            "description": "Adversaries may send spearphishing emails with a malicious link in an attempt to gain access to victim systems. Spearphishing with a link is a specific variant of spearphishing. It is different from other forms of spearphishing in that it employs the use of links to download malware contained in email, instead of attaching malicious files to the email itself, to avoid defenses that may inspect email attachments. \n\nAll forms of spearphishing are electronically delivered social engineering targeted at a specific individual, company, or industry. In this case, the malicious emails contain links. Generally, the links will be accompanied by social engineering text and require the user to actively click or copy and paste a URL into a browser, leveraging [User Execution](https://attack.mitre.org/techniques/T1204). The visited website may compromise the web browser using an exploit, or the user will be prompted to download applications, documents, zip files, or even executables depending on the pretext for the email in the first place. Adversaries may also include links that are intended to interact directly with an email reader, including embedded images intended to exploit the end system directly or verify the receipt of an email (i.e. web bugs/web beacons). Links may also direct users to malicious applications designed to [Steal Application Access Token](https://attack.mitre.org/techniques/T1528)s, like OAuth tokens, in order to gain access to protected applications and information.(Citation: Trend Micro Pawn Storm OAuth 2017)",
            "id": "attack-pattern--6cd1a813-ccdf-4ba0-9b54-cb808f1059cc",
            "name": "Phishing: Spearphishing Link",
            "lang": "en",
            "created": "2019-10-24T12:51:37.147416Z",
            "modified": "2021-04-13T16:01:41.879086Z",
            "kill_chain_phases": [
                {
                    "kill_chain_name": "mitre-attack",
                    "phase_name": "initial-access"
                }
            ],
            "created_by_ref": "identity--357447d7-9229-4ce1-b7fa-f1b83587048e",
            "revoked": false,
            "x_mitre_contributors": [
                "Shailesh Tiwary (Indian Army)",
                "Mark Wee",
                "Jeff Sakowicz, Microsoft Identity Developer Platform Services (IDPM Services)",
                "Saisha Agrawal, Microsoft Threat Intelligent Center (MSTIC)"
            ],
            "x_mitre_data_sources": [
                "Packet capture",
                "Web proxy",
                "Email gateway",
                "Detonation chamber",
                "SSL/TLS inspection",
                "DNS records",
                "Mail server"
            ],
            "x_mitre_detection": "URL inspection within email (including expanding shortened links) can help detect links leading to known malicious sites. Detonation chambers can be used to detect these links and either automatically go to these sites to determine if they're potentially malicious, or wait and capture the content if a user visits the link.\n\nBecause this technique usually involves user interaction on the endpoint, many of the possible detections take place once [User Execution](https://attack.mitre.org/techniques/T1204) occurs.",
            "x_mitre_is_subtechnique": true,
            "x_mitre_platforms": [
                "Linux",
                "macOS",
                "Windows",
                "Office 365",
                "SaaS"
            ],
            "x_mitre_version": "2.0",
            "nb_deps": 4
        }
    ]
}
```

**Comment by SamuelHassine** [comment]
Hello @Darkheir,

This will be definitely fixed in 4.5.1 (next release).

Kind regards,
Samuel

**Comment by Darkheir** [comment]
Awesome! 🙂
